### DISCUSSION TITLE
BIRT  splitting report into 3 pages (3 sheets) )using the genreport.bat

### DISCUSSION THREAD

**Discussion by Anka93** [eclipse-birt/birt#1202] [Q&A] [answered]
I form an Excel file through genreport.bat.
my report has 3 pages, but after genreport.bat the report is displayed on one page in excel.
how to achieve splitting report into 3 pages (3 sheets) )using the genreport.bat ?

**Comment by hvbtup** [comment]
The genreport.bat script is only useful for testing purposes.
The reason is that BIRT startup is slow and with this script you have to startup BIRT anew for every single report.
If you want to use BIRT in production, you should either integrate BIRT into your Java application or (independent from the programming-language) use the viewer servlet (birt.war) to run reports via http.

So the question *should* be: How to achieve this effect with the viewer servlet?
I cannot answer this question, but maybe someone else can?

**Reply by Anka93** [reply]
Oh no.
my task is to generate a report so that it is automatically downloaded to disk.
which is why I need to use genreport.bat

**Reply by hvbtup** [reply]
But this is also possible with the other two ways of calling BIRT I described.

**Reply by Anka93** [reply]
I am quite new to birt.
tell me how to download the report in xls automatically(or with scripts) when using birt.war?

**Reply by hvbtup** [reply]
To automate generating and downloading reports with the viewer, use any http client.
If you are on the command line, consider using `curl`.

**Reply by hvbtup** [reply]
BTW see the BIRT web site pages https://eclipse.github.io/birt-website/docs/integrating/viewer-setup and https://eclipse.github.io/birt-website/docs/integrating/viewer-usage for more information about using the viewer.

**Reply by hvbtup** [reply]
Regarding the page-breaking questions, please be more specific.
There are two reasons for a page-break:
A manual page-break (which you can define using the page-break-before or page-break-after properties of a layout item or a group) and automatic page-breaks that occur due to the limited size of a page. Since a spread sheet is unlimited, these automatic page-breaks don't make sense for Excel output.

**Reply by wimjongman** [reply]
I think that the question is how to create three workbooks in one excel. AFAIK, this cannot be done in BIRT.

**Reply by hvbtup** [reply]
You mean worksheets. This can be done with the Spudsoft emitter, at least with our modified fork (tup_main branch at triestram-partner/birt), see attachment.
[excel_sheets.zip](https://github.com/eclipse/birt/files/10673693/excel_sheets.zip)

**Reply by wimjongman** [reply]
Yes, worksheets. That is cool! Do you do this based on a page break?

**Reply by hvbtup** [reply]
Yes, see the list group page-break properties.

**Reply by hvbtup** [reply]
@wimjongman I feel a little bit proud for knowing something about BIRT which you didn't know 😃

**Reply by wimjongman** [reply]
LOL. Trust me, when it comes to the actual usage of BIRT, you know  _a lot_ more than me.

**Comment by wimjongman** [comment, marked as answer]
So getting back to @Anka93's original question: She/He/Etc.. can try the report design in your attached zip and look at the properties of list-group to see how it can be achieved?

**Reply by hvbtup** [reply]
Yes. Straightforward. But remember this output was created with my slightly different version of BIRT. I don't remember if all the latest changes from Jim Talbut (aka jaytay) actually made it into the master branch. Can you just test this yourself?

**Reply by wimjongman** [reply]
I tested your report on vanilla BIRT, and it works like a charm.

Is it just any page break on a group change that causes this, or is there more too it?

<details><summary>Image of group properties</summary>

![image](https://user-images.githubusercontent.com/180969/217291755-2508d09a-3e38-4b8c-92be-0c96c37e7fb4.png)

</details>

**Reply by hvbtup** [reply]
Yes and yes :-)

AFAIK a forced page break is usually causing a "sheet-break" in the Spudsoft emitter.

But this can probably be changed - the emitter is highly configurable with render options and user properties.
Unfortunately the documentation that was available online on bitbucket.org is no longer available, so for more advanced usage the workflow is: dig into the code and look for interesting options and user properties, than use them in a report and see what happens.

There's more to it, because the way that the sheet names are generated is coming from the TOC entries, AFAIK.
So it's easiest to use grouping with TOC entries to define the sheet names.
